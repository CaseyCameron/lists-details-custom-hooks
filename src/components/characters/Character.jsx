import React from 'react';

const Character = ({ name, affiliation, photoUrl }) => {
  return (
    <figure>
      <figcaption>
        <p>{name}</p>
      </figcaption>
      <img src={photoUrl} alt={name} />
    </figure>
  )
};

export default Character;