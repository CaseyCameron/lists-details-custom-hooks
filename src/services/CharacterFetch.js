export const fetchCharacters = async (page) => {
  const res = await fetch(
    `https://last-airbender-api.herokuapp.com/api/v1/characters?perPage=10&page=${page}`
  );
  const characters = await res.json();

  return characters;
};

export const fetchCharacterById = async (id) => {
  const res = await fetch(`https://last-airbender-api.herokuapp.com/api/v1/characters/${id}`);

  return res.json();
};