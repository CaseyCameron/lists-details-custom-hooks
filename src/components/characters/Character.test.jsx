import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { AvatarCharacters } from '../../services/MockData';
import Character from './Character';

const server = setupServer(
  rest.get('https://last-airbender-api.herokuapp.com/api/v1/characters/5cf5679a915ecad153ab68d1', (req, res, ctx) => {
    return res(ctx.json(AvatarCharacters[1]))
  })
);

describe.only('Avatar characters list', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('renders a character on the home page', () => {
    const { asFragment } = render(
      <Character
        name='Analay'
        affiliation='Unnamed team'
        photoUrl='https://vignette.wikia.nocookie.net/avatar/images/c/c3/Analay.png/revision/latest?cb=20140430182201'
      />
    )
    expect(asFragment()).toMatchSnapshot();
  });
});