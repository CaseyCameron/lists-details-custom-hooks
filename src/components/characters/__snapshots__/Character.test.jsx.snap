// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a character on the home page 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="Analay"
      src="https://vignette.wikia.nocookie.net/avatar/images/c/c3/Analay.png/revision/latest?cb=20140430182201"
    />
    <figcaption>
      <p>
        Analay
      </p>
      <p>
        Unnamed team
      </p>
      <p />
    </figcaption>
  </figure>
</DocumentFragment>
`;
